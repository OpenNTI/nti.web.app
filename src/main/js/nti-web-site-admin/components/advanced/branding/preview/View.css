.preview-dialog {
	width: 768px;
}

.preview-dialog ul {
	list-style: none;
	padding: 0;
	margin: 0;
}

.preview-dialog > ul > li {
	box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.15);
	margin-bottom: 2rem;
}

.title {
	color: white;
	font-size: 1.125rem;
	margin: 0.5em 0;
}

.preview-dialog :global(.icon-light-x) {
	background: white;
	border-radius: 50%;
	font-size: 1.75rem;
	padding: 0.75rem;
	position: absolute;
	right: calc(100% + 2rem);
	cursor: pointer;
}

.controls {
	text-align: center;
}
