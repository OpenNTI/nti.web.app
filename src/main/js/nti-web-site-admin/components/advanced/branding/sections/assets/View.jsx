import React from 'react';
import PropTypes from 'prop-types';
import {scoped} from '@nti/lib-locale';

import Section from '../Section';

import {types} from './constants';
import AssetItem from './AssetItem';

const t = scoped('nti-web-app.admin.branding.logo', {
	title: 'Logo Assets',
	types: {
		[types.icon]: {
			title: 'Square Format',
			description: 'We recommend this asset be close to a square in proportion.'
		},
		[types.logo]: {
			title: 'Logo',
			description: 'The wider orientation has space for your icon and company name.'
		},
		[types.fullLogo]: {
			title: 'Wide Format',
			description: 'The wider orientation has space for your icon and company name.'
		},
		[types.email]: {
			title: 'Email Header',
			description: 'The email template is flexibile enough to accomodate various lockups.'
		},
		[types.favicon]: {
			title: 'Favicon',
			description: 'This square asset helps users identify your site among their open browser tabs.'
		},
		// [types.login]: {
		// 	title: 'Login Image',
		// 	description: 'Welcome people back with an image that speaks to your brand.'
		// },
	}
});

export default function Assets ({assets = {}, onChange}) {

	const changeHandler = type => item => onChange(type, item);
	const getSrc = type => (assets[type] || {}).source;

	return (
		<Section text={t}>
			{
				Object.values(types).map(type => (
					<AssetItem
						key={type}
						name={type}
						src={getSrc(type)}
						onChange={changeHandler(type)}
						getText={k => t(['types', type, k])}
					/>
				))
			}
		</Section>
	);
}

Assets.propTypes = {
	assets: PropTypes.object,
	onChange: PropTypes.func.isRequired
};
