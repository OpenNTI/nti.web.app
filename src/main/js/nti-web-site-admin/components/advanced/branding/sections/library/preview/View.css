.preview-root {
	background: grey;
	height: 480px;
	margin-right: -2rem; /* counter container padding to get flush right edge */
}
