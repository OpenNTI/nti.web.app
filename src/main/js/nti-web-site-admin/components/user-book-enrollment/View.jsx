import {Router, Route} from '@nti/web-routing';

import Overview from './overview';
import Reports from './reports';
import Frame from './Frame';

export default Router.for([
	Route({path: '/reports', component: Reports, name: 'site-admin.users.user-book.reports'}),
	Route({path: '/', component: Overview, name: 'site-admin.users.user-book.overview'})
], {frame: Frame});
