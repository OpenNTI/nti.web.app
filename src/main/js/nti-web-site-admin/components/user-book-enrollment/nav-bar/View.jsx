import React from 'react';

import Card from '../../common/Card';

import Identity from './Identity';
import Tabs from './Tabs';

export default function SiteAdminUserBookIdentity (props) {
	return (
		<Card className="site-admin-user-enrollment-nav-bar">
			<Identity {...props} />
			<Tabs {...props} />
		</Card>
	);
}
