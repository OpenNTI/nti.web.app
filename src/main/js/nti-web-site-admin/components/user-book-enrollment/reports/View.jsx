import React from 'react';
import PropTypes from 'prop-types';
import {List} from '@nti/web-reports';

import Card from '../../common/Card';

SiteAdminUserEnrollmentReports.propTypes = {
	userBookRecord: PropTypes.object
};
export default function SiteAdminUserEnrollmentReports ({userBookRecord}) {
	return userBookRecord ?
		(<Card><List context={userBookRecord} /></Card>) :
		null;
}
