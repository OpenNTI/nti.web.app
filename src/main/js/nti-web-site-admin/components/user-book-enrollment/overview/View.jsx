import './View.scss';
import React from 'react';
import PropTypes from 'prop-types';
import { scoped } from '@nti/lib-locale';
import { DateTime } from '@nti/web-commons';

import DateValue from '../../common/DateValue';

const t = scoped(
	'nti-web-site-admin.components.user-book-enrollment.Overview',
	{
		lastSeen: 'Last Active',
	}
);

export default class SiteAdminUserBookView extends React.Component {
	static propTypes = {
		userBookRecord: PropTypes.object,
	};

	render() {
		const { userBookRecord } = this.props;

		return (
			<div className="site-admin-user-enrollment-overview">
				{userBookRecord && (
					<DateValue
						date={this.props.userBookRecord.getLastSeenTime()}
						format={DateTime.MONTH_NAME_DAY_YEAR_TIME}
						label={t('lastSeen')}
					/>
				)}
			</div>
		);
	}
}
