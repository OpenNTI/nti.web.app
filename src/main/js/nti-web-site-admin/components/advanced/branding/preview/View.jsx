import React from 'react';
import PropTypes from 'prop-types';
import {Button, Prompt} from '@nti/web-commons';
import classnames from 'classnames/bind';

import t from '../sections/assets/strings';
import {types} from '../sections/assets/constants';

import getWidget from './widgets';
import styles from './View.css';

const cx = classnames.bind(styles);

export default function Preview ({onSave, onClose}) {
	return (
		<Prompt.Dialog>
			<div className={cx('preview-dialog')}>
				<span className="icon-light-x" onClick={onClose} />
				<ul>
					{
						Object.values(types).map(type => {
							const Widget = getWidget(type);
							return !Widget ? null : (
								<li key={type}>
									<div className={cx('title')}>{t(['types', type, 'title'])}</div>
									<Widget />
								</li>
							);
						})
					}
				</ul>
				{ onSave && (
					<div className={cx('controls')}>
						<Button onClick={onSave}>Apply Changes</Button>
					</div>
				)}
			</div>
		</Prompt.Dialog>
	);
}
