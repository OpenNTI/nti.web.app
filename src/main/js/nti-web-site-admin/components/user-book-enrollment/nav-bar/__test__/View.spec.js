import React from 'react';
import renderer from 'react-test-renderer';

import { TestUtils } from '@nti/web-client';

import View from '../View';

const { tearDownTestClient, setupTestClient } = TestUtils;

const getMockService = () => {
	return {};
};

const onBefore = () => {
	jest.useFakeTimers();
	setupTestClient(getMockService());
};

const onAfter = () => {
	tearDownTestClient();
};

/* eslint-env jest */
describe('Site admin user book enrollment nav bar test', () => {
	beforeEach(() => onBefore());
	afterEach(onAfter);

	test('Basic render test', async () => {
		const user = {
			Username: 'testUser',
		};

		const book = {
			title: 'book1',
			getDefaultAssetRoot() {
				return 'testRoot';
			},
		};

		const cmp = renderer.create(<View user={user} book={book} />);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
