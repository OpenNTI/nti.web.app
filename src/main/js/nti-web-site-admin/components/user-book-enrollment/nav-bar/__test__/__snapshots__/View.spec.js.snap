// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin user book enrollment nav bar test Basic render test 1`] = `
<div
  className="site-admin-card site-admin-user-enrollment-nav-bar"
>
  <div
    className="site-admin-user-enrollment-identity"
  >
    <div
      className="profile-image"
    >
      <img
        className="enrollment-course-image"
        onError={[Function]}
        src="contentpackage-landing-232x170.png?t=0"
      />
      <div
        className="user-avatar"
      />
    </div>
    <div
      className="enrollment-info"
    >
      <div
        className="plain-text"
      >
        Activity for
      </div>
      <div
        className="plain-text"
      >
        in
      </div>
      <div
        className="course-title"
      >
        book1
      </div>
    </div>
  </div>
  <ul
    className="unadorned-list site-admin-user-tabs"
  >
    <li>
      <div
        className="nti-link-to-path"
        exact={true}
        to="./"
      >
        Overview
      </div>
    </li>
    <li>
      <div
        className="nti-link-to-path"
        to="./reports"
      >
        Reports
      </div>
    </li>
  </ul>
</div>
`;
